### PR TITLE
feat: New logs page

### PR DESCRIPTION
**Pull request by ogzhanolguncu**
## What does this PR do?

Fixes # (issue)

*If there is not an issue for this, please create one first. This is used to tracking purposes and also helps use understand why this PR exists*

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- Test A
- Test B

### Required

- [ ] Filled out the "How to test" section in this PR
- [ ] Read [Contributing Guide](./CONTRIBUTING.md)
- [ ] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand areas
- [ ] Ran `pnpm build`
- [ ] Ran `pnpm fmt`
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Unkey Docs if changes were necessary

## Summary by CodeRabbit

- **New Features**
	- Introduced a `LogsChart` component to visualize log data with a bar chart.
	- Added a `DatePickerWithRange` component for selecting date ranges.
	- Implemented a `ResponseStatus` component for filtering logs by HTTP response status codes.
	- Added a `SearchCombobox` for managing search items.
	- Enhanced log filtering with a new `LogsFilters` component.
	- Introduced `LogDetails` components for displaying detailed log information.
	- Added a `TimestampInfo` component for formatting and displaying timestamps.
	- Introduced a `Calendar` component for date selection.
	- Implemented a `LogsTable` component for displaying logs in a virtualized format.
	- Added a `ResizablePanel` component for adjustable log detail views.
	- Implemented a `ButtonGroup` component for grouping buttons.
	- Enhanced the `CommandItem` component to support a disabled state.
	- Added a `LogMetaSection` component for displaying meta information about logs.
	- Introduced a `LogSection` component for organized display of log details.

- **Bug Fixes**
	- Improved error handling for log fetching and querying processes.

- **Documentation**
	- Updated documentation to reflect new components and functionalities.

- **Chores**
	- Added new dependencies to enhance data visualization and user interaction capabilities.

### CODE REVIEW

File: apps/dashboard/app/(app)/logs/components/chart.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    aggregatedData.push({
+      date: format(timestamp, "yyyy-MM-dd'T'HH:mm:ss"),
+      success,
+      warning,
+      error,
+    });
+  }
+
+  return aggregatedData;
+}
```
[the remark this answers is not in the document]
Comment (chronark) [reply apps/dashboard/app/(app)/logs/components/chart.tsx:79]: yeah that feels a bit better, our arrays aren't huge, but as we might add more status types, we could end up looping over them quite a few times.

File: apps/dashboard/components/ui/group-button.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    <div
+      className={cn(
+        "flex",
+        {
+          "flex-col": isVertical,
+          "w-fit": isVertical,
+        },
+        className,
+      )}
+    >
```
[the remark this answers is not in the document]
Comment (ogzhanolguncu) [reply apps/dashboard/components/ui/group-button.tsx:31]: there is no `role` called "group"

File: apps/dashboard/app/(app)/logs/components/chart.tsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            return date.toLocaleDateString("en-US", {
+              month: "short",
+              day: "2-digit",
+              hour: "2-digit",
+              minute: "2-digit",
+              hour12: true,
+            });
+          }}
+        />
+        <CartesianGrid strokeDasharray="2" stroke="#DFE2E6" vertical={false} />
```
Comment (chronark) [inline apps/dashboard/app/(app)/logs/components/chart.tsx:103]: how will this work in dark/light mode?
can we reference a css variable instead so it works automatically?

File: apps/dashboard/app/(app)/logs/components/filters/components/custom-date-filter.tsx [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+import { useLogSearchParams } from "../../../query-state";
+import TimeSplitInput from "./time-split";
+
+export function DatePickerWithRange({ className }: React.HTMLAttributes<HTMLDivElement>) {
+  const [interimDate, setInterimDate] = useState<DateRange>({
+    from: new Date(),
+    to: new Date(),
+  });
+  const [finalDate, setFinalDate] = useState<DateRange>();
+  const [startTime, setStartTime] = useState({ HH: "09", mm: "00", ss: "00" });
```
Comment (chronark) [inline apps/dashboard/app/(app)/logs/components/filters/components/custom-date-filter.tsx:21]: where do these defaults come from?

File: apps/dashboard/app/(app)/logs/components/filters/components/response-status.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+interface CheckboxItemProps {
+  id: string;
+  label: string;
+  description: string;
+  checked: boolean;
+  onCheckedChange: (checked: boolean) => void;
+}
+
+const checkboxItems = [
+  { id: "500", label: "Error", description: "500 error codes" },
```
Comment (chronark) [inline apps/dashboard/app/(app)/logs/components/filters/components/response-status.tsx:15]: let's go with "5xx error codes" etc, to indicate that it's not just 500, but also 503 etc

File: apps/dashboard/app/(app)/logs/components/log-details/components/log-footer.tsx [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+          tooltipContent: "Copy Request User Agent",
+          tooltipSuccessMessage: "Request user agent copied to clipboard",
+        },
+        {
+          label: "Outcome",
+          description: (content) => {
+            let contentCopy = content;
+            if (contentCopy == null) {
+              contentCopy = "VALID";
+            }
```
[the remark this answers is not in the document]
Comment (chronark) [reply apps/dashboard/app/(app)/logs/components/log-details/components/log-footer.tsx:61]: can we use something like "N/A" or anything else?
labeling this as valid as a fallback might end up confusing a customer later

File: apps/dashboard/app/(app)/logs/components/log-details/components/log-header.tsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  log: Log;
+  onClose: () => void;
+};
+export const LogHeader = ({ onClose, log }: Props) => {
+  return (
+    <div className="border-b-[1px] px-3 py-4 flex justify-between border-border items-center">
+      <div className="flex gap-2">
+        <Badge variant="secondary" className="bg-transparent">
+          POST
+        </Badge>
```
[the remark this answers is not in the document]
Comment (chronark) [reply apps/dashboard/app/(app)/logs/components/log-details/components/log-header.tsx:17]: does the log have a `.method`? in that case I agree
generally we only use GET for reads and POST for any mutations, we don't use PUT/DELETE etc

File: apps/dashboard/app/(app)/logs/components/log-details/components/meta-content.tsx [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+      });
+      setHtml(html);
+    });
+  }, [content]);
+
+  return (
+    <Card className="rounded-[5px]">
+      <CardContent
+        className="whitespace-pre-wrap text-[12px] w-[300px]"
+        dangerouslySetInnerHTML={{
```
Comment (chronark) [inline apps/dashboard/app/(app)/logs/components/log-details/components/meta-content.tsx:31]: why do we need to use this?

File: apps/dashboard/app/(app)/logs/logs-page.tsx [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+    updateLogs();
+  }, [updateLogs]);
+
+  return (
+    <div className="flex flex-col gap-4 items-start w-full overflow-y-hidden">
+      <LogsFilters />
+      <LogsChart logs={logs} />
+      <LogsTable logs={logs} />
+    </div>
+  );
```
[the remark this answers is not in the document]
Comment (ogzhanolguncu) [reply apps/dashboard/app/(app)/logs/logs-page.tsx:83]: Already did in <LogsTable />

File: apps/dashboard/app/(app)/logs/components/log-details/index.tsx [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+      onClose={onClose}
+      className="absolute right-0 bg-background border-l border-t border-solid font-mono border-border shadow-md overflow-y-auto z-[3]"
+      style={panelStyle}
+    >
+      <LogHeader log={log} onClose={onClose} />
+
+      <div className="space-y-3 border-b-[1px] border-border py-4">
+        <div className="mt-[-24px]" />
+        <Headers headers={log.request_headers} title="Request Header" />
+        <Headers headers={flattenObject(JSON.parse(log.request_body))} title="Request Body" />
```
Comment (chronark) [inline apps/dashboard/app/(app)/logs/components/log-details/index.tsx:52]: this naming is weird, it's called "Headers" but then receives the request body oO

File: apps/dashboard/app/(app)/logs/logs-page.tsx [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+
+  const updateLogs = useCallback(() => {
+    // If any filter is set, replace all logs with new data
+    const hasFilters =
+      !searchParams.host ||
+      !searchParams.requestId ||
+      !searchParams.path ||
+      !searchParams.method ||
+      !searchParams.responseStatus;
```
[the remark this answers is not in the document]
Comment (chronark) [reply apps/dashboard/app/(app)/logs/logs-page.tsx:51]: yeah I don't get it either

File: apps/dashboard/app/(app)/logs/components/log-details/resizable-panel.tsx [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+        return;
+      }
+
+      const containerRect = panelRef.current.getBoundingClientRect();
+      const newWidth = Math.min(Math.max(containerRect.right - e.clientX, minW), maxW);
+      setWidth(`${newWidth}px`);
+      onResize?.(newWidth);
+    },
+    [isDragging, minW, maxW, onResize],
+  );
```
[the remark this answers is not in the document]
Comment (ogzhanolguncu) [reply apps/dashboard/app/(app)/logs/components/log-details/resizable-panel.tsx:49]: Resize is already debounced

File: apps/dashboard/app/(app)/logs/components/log-details/components/meta-content.tsx (hunk 8 above)
Comment (ogzhanolguncu) [reply apps/dashboard/app/(app)/logs/components/log-details/components/meta-content.tsx:31]: Code formatted by "shiki" doesn't work if not injected like this. But, i'll double check anyways. We might have to purify them just incase.

File: apps/dashboard/app/(app)/logs/components/filters/components/custom-date-filter.tsx (hunk 4 above)
Comment (ogzhanolguncu) [reply apps/dashboard/app/(app)/logs/components/filters/components/custom-date-filter.tsx:21]: It's just a random starting hour. Do you have something else on your mind?

File: apps/dashboard/app/(app)/logs/components/log-details/index.tsx (hunk 10 above)
Comment (ogzhanolguncu) [reply apps/dashboard/app/(app)/logs/components/log-details/index.tsx:52]: When I first created that component it was only handling headers than evolved into something else 😄. Fair, I'll fix the name.

File: apps/dashboard/app/(app)/logs/components/log-details/components/meta-content.tsx (hunk 8 above)
Comment (chronark) [reply apps/dashboard/app/(app)/logs/components/log-details/components/meta-content.tsx:31]: why do we need shiki anyway?
what's wrong with a good old `pre` tag?

Comment (ogzhanolguncu) [reply apps/dashboard/app/(app)/logs/components/log-details/components/meta-content.tsx:31]: Yeah, highlighting doesn't have any benefits for our meta fields. Initially I was using this for every jsonish field for log details.

**Comment by ogzhanolguncu** [comment]
Moved to here due to too many lock conflicts -> https://github.com/unkeyed/unkey/compare/logs-page-v2

**Pull request closed without merging**